source/s3_client.c

```diff
@@ -1515,6 +1514,12 @@ void aws_s3_client_update_connections_threaded(struct aws_s3_client *client) {
 
         struct aws_s3_request *request = aws_s3_client_dequeue_request_threaded(client);
         const uint32_t max_active_connections = aws_s3_client_get_max_active_connections(client, request->meta_request);
+        if (request->is_noop) {
+            /* If request is no-op, finishes and cleans up the request */
+            s_s3_client_meta_request_finished_request(client, request->meta_request, request, AWS_ERROR_SUCCESS);
+            request = aws_s3_request_release(request);
+            continue;
+        }
 
         /* Unless the request is marked "always send", if this meta request has a finish result, then finish the request
          * now and release it. */
```

Comment:
trivial, but the code after this is an `if/else if/else if/else` sequence. May as well insert this block into the sequence. vs a standalone `if (no-op) continue` followed by a `if/else if/else if/else` sequence
```suggestion
        } else if (!request->always_send && aws_s3_meta_request_has_finish_result(request->meta_request)) {
            /* Unless the request is marked "always send", if this meta request has a finish result,
             * then finish the request now and release it. */
```